### PR TITLE
MWPW-147905-Move image span value outside of alt value

### PR DESCRIPTION
**Pull request by Ruchika4**
* Move image span value outside of alt value

Resolves: [MWPW-147905](https://jira.corp.adobe.com/browse/MWPW-147905)

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.live/drafts/ruchika/firefly/take2/marquee8?martech=off
- After: https://image-span--cc--adobecom.hlx.live/drafts/ruchika/firefly/take2/marquee8?martech=off

### CODE REVIEW

**Comment by spadmasa** [comment]
Validation done on the below url on new authoring change :
https://image-span--cc--adobecom.hlx.page/creativecloud/animation/marquee61
image

image

Verified the localization flow on docx and html flow
https://image-span--cc--adobecom.hlx.page/il_he/creativecloud/animation/marquee61
image
https://image-span--cc--adobecom.hlx.page/de/creativecloud/animation/marquee6

**Comment by aishwaryamathuria** [comment]
@Ruchika4 Can we remove the icon symbol ":span-n:" and use "span-n" with dnt if we do not explicitly need the icons?
"::" will call additional processing to replace these with svg which we do not need and we can avoid that?
Would the processing be easier if you just add it in an ordered list? With span image and prompt in each list item?

@salonijain3 @amitbikram Could you share your inputs as well? If we should sort out the content in the foregrould display section?

File: creativecloud/features/firefly/firefly-masonry.js [hunk 1]
```diff
@@ -182,15 +183,12 @@ export default async function setMultiImageMarquee(el) {
...
-        alt = '';
-        span = s.querySelector('img').getAttribute('alt');
-      }
+      const alt = s.querySelector('img').getAttribute('alt');
+      const span = allP[[...allP].indexOf(s) - 1].querySelector('span')?.classList[1]?.split('icon-');
       mediaDetail.imgSrc.push(src);
       mediaDetail.prompt.push(prompt);
       mediaDetail.href.push(href);
-      mediaDetail.spans.push(span);
+      if (span?.length > 1)mediaDetail.spans.push(span[1]);
```
Comment (aishwaryamathuria) [inline creativecloud/features/firefly/firefly-masonry.js:191]:
```suggestion
      if (span?.length > 1) mediaDetail.spans.push(span[1]);
```

**Review by aishwaryamathuria — CHANGES_REQUESTED** [review]

**Comment by Ruchika4** [comment]
> @Ruchika4
> 
> 1. Can we remove the icon symbol ":span-n:" and use "span-n" with dnt if we do not explicitly need the icons?
>    "::" will call additional processing to replace these with svg which we do not need and we can avoid that?
> 2. Would the processing be easier if you just add each item to be displayed on masonry layout it in an ordered list? With span image and prompt in each list item? **Or**
>    have an ordered list at the end with each list item containing the ordering in the row? Like following:
> 
> ```
> 1. span-8 span-4
> 2. span-4 span-4 span-4
> ```
> 
> 1. @salonijain3 @amitbikram Could you share your inputs as well? If we should sort out the content in the foregrould display section?

Hi @aishwaryamathuria , we cannot do the first approach as it will not work in html localization workflow and span value will get localized. In html work flow DNT style will not work. Same with second approach too span value will get translated as per my understanding. Tagging @janaki-r-bhagwath to help confirm.

File: creativecloud/features/firefly/firefly-masonry.js [hunk 2]
```diff
@@ -87,7 +87,8 @@ function processMasonryMedia(gridDiv, allP, mediaDetail) {
   handleTouchDevice(mediaContainer, 2000);
 
   const spanWidth = mediaDetail.spans[lastIndex];
-  mediaContainer.classList.add(`ff-grid-${spanWidth.trim().replace(' ', '-')}`);
+  const spanClass = spanWidth === undefined ? 'span-4' : `${spanWidth?.trim().replace(' ', '-')}`;
```
Comment (aishwaryamathuria) [inline creativecloud/features/firefly/firefly-masonry.js:90]:
```suggestion
  const spanClass = spanWidth ? 'span-4' : `${spanWidth?.trim().replace(' ', '-')}`;
```

File: creativecloud/features/firefly/firefly-masonry.js [hunk 3]
```diff
@@ -170,30 +171,37 @@ export default async function setMultiImageMarquee(el) {
   const ic = el.querySelector('.interactive-container');
   const mediaElements = el.querySelector('.asset');
   const allP = mediaElements.querySelectorAll('p:not(:empty)');
+  const allLi = el.querySelectorAll('li');
```
Comment (aishwaryamathuria) [inline creativecloud/features/firefly/firefly-masonry.js:174]:
```suggestion
  const allLi = el.querySelectorAll('.assets li');
```

Comment (aishwaryamathuria) [reply creativecloud/features/firefly/firefly-masonry.js:174]: The text area can have product tiles of list as well

**Review by aishwaryamathuria — CHANGES_REQUESTED** [review]

**Review by aishwaryamathuria — APPROVED** [review]

File: creativecloud/features/firefly/firefly-masonry.js (hunk 2 above)
Comment (Ruchika4) [reply creativecloud/features/firefly/firefly-masonry.js:90]: Thanks for the review @aishwaryamathuria. I believe above will set the span-4 for all images even when it's defined in word doc.
I can use the below code. please let me know if you see any concern with that.
`const spanClass = spanWidth ? `${spanWidth?.trim().replace(' ', '-')}` : 'span-4';`

**Review by drashti1712 — APPROVED** [review]

**Comment by spadmasa** [comment]
Verified the flow in the below urls 

authoring updated :
https://image-span--cc--adobecom.hlx.live/drafts/ruchika/firefly/take2/marquee8?martech=off

**Pull request merged**